### ISSUE TITLE
Bug with pop turret alike

### ISSUE THREAD

**Issue by dnqbob** [OpenRA/OpenRA#18793]
![1](https://user-images.githubusercontent.com/13763394/97775567-b6c2cd00-1b9c-11eb-93a3-aa78d8963c53.gif)

![2](https://user-images.githubusercontent.com/13763394/97775570-b88c9080-1b9c-11eb-992b-fae3567bf458.gif)

**Comment by dnqbob** [comment]
1. bug with damaged turret, it is not turning
2. the turret should hiding when build up.

**Comment by abcdefg30** [comment]
Which version is that on? I can't reproduce 1. and 2. is intentional afaik.

**Comment by dnqbob** [comment]
I use the newest "bleed" branch which just after a pull today

**Comment by dnqbob** [comment]
The head commit
<img width="282" alt="1" src="https://user-images.githubusercontent.com/13763394/97781741-61041a00-1bc8-11eb-9957-0b94a2faadf4.png">

bug again (tester need to reduce the health of Samsite below 50% for this bug)
![1](https://user-images.githubusercontent.com/13763394/97781742-63667400-1bc8-11eb-8fb2-37af6a085c7e.gif)

**Comment by dnqbob** [comment]
if 2 is a regular setting then, it is OK

**Comment by reaperrr** [comment]
1 is the same bug the other turrets used to have before https://github.com/OpenRA/OpenRA/pull/18738

Basically, something goes haywire in the turret initialization if it's damaged "too early".

**Comment by pchote** [comment]
Are we sure that bug isn't a new regression? #18738 didn't fix the underlying issue, just moved turrets over to other traits that don't suffer it.

**Comment by reaperrr** [comment]
> Are we sure that bug isn't a new regression?

Not really. I just couldn't find anything obviously wrong last time I tried to investigate this, and there have been similar issues in the past that may or may not be caused by the same bug/core issue.

**Comment by dnqbob** [comment]
![动画](https://user-images.githubusercontent.com/13763394/137947959-cb67f3a5-6a02-471f-94ef-79dcbe840062.gif)

A new report in our mod community
